neuralop/layers/skip_connections.py

```diff
@@ -14,25 +23,43 @@ def skip_connection(in_features, out_features, n_dim=2, bias=False, type="soft-g
         number of output features
     n_dim : int, default is 2
         Dimensionality of the input (excluding batch-size and channels).
-        ``n_dim=2`` corresponds to having Module2D. 
+        ``n_dim=2`` corresponds to having Module2D.
     bias : bool, optional
         whether to use a bias, by default False
-    type : {'identity', 'linear', soft-gating'}
+    skip_type : {'identity', 'linear', soft-gating'}
         kind of skip connection to use, by default "soft-gating"
+    kwargs :  Dict[str, Any], optional
+        Args to pass to connections (ex. ``Conv`` in case of "linear", etc).
+        Args include ``"device"``, ``"dtype"``.
 
     Returns
     -------
     nn.Module
         module that takes in x and returns skip(x)
     """
-    if type.lower() == 'soft-gating':
-        return SoftGating(in_features=in_features, out_features=out_features, bias=bias, n_dim=n_dim)
-    elif type.lower() == 'linear':
-        return getattr(nn, f'Conv{n_dim}d')(in_channels=in_features, out_channels=out_features, kernel_size=1, bias=bias)
-    elif type.lower() == 'identity':
+    if skip_type.lower() == 'soft-gating':
+        return SoftGating(
+            in_features=in_features,
+            out_features=out_features,
+            bias=bias,
+            n_dim=n_dim,
+        )
+    elif skip_type.lower() == 'linear':
+        Conv = getattr(nn, f'Conv{n_dim}d')
+        return Conv(
+            in_channels=in_features,
+            out_channels=out_features,
+            kernel_size=1,
+            bias=bias,
+            device=kwargs.get('device'),
```

Comment:
Same here let's make it explicit, the kwargs is more meant to silently absorb arguments not-common to all the skips but not the best pattern either.